databroker/src/broker.rs

```diff
@@ -1530,6 +1592,278 @@ impl<'a, 'b> AuthorizedAccess<'a, 'b> {
+                        }
+                        Err(PermissionError::Expired) => {
+                            let message = "Permission expired".to_string();
+                            return Err((ActuationError::PermissionExpired, message));
+                        }
+                    }
+                }
+                Err(ReadError::NotFound) => {
+                    let message = format!("Could not resolve vss_path of vss_id {}", vss_id);
+                    return Err((ActuationError::NotFound, message));
+                }
+                Err(ReadError::PermissionDenied) => {
+                    let message = format!("Permission denied for vss_id {}", vss_id);
+                    return Err((ActuationError::PermissionDenied, message));
+                }
+                Err(ReadError::PermissionExpired) => {
+                    return Err((
+                        ActuationError::PermissionExpired,
+                        "Permission expired".to_string(),
+                    ))
+                }
+            }
+        }
+
+        let actuation_changes_per_vss_id = &self
+            .map_actuation_changes_by_vss_id(actuation_changes)
+            .await;
+        for actuation_change_per_vss_id in actuation_changes_per_vss_id {
+            let vss_id = *actuation_change_per_vss_id.0;
+            let actuation_changes = actuation_change_per_vss_id.1.clone();
+
+            let opt_actuation_subscription = actuation_subscriptions
+                .iter()
+                .find(|subscription| subscription.vss_ids.contains(&vss_id));
+            match opt_actuation_subscription {
+                Some(actuation_subscription) => {
+                    let is_expired = actuation_subscription.permissions.expired();
+                    match is_expired {
+                        Err(_) => {
+                            let message = format!(
+                                "Permission for vss_ids {:?} expired",
+                                actuation_subscription.vss_ids
+                            );
+                            return Err((ActuationError::PermissionExpired, message));
+                        }
+                        Ok(_) => {
+                            if !actuation_subscription.actuation_provider.is_available() {
+                                let message =
+                                    format!("Provider for vss_id {} does not exist", vss_id);
+                                return Err((ActuationError::ProviderNotAvailable, message));
+                            }
+
+                            actuation_subscription
+                                .actuation_provider
+                                .actuate(actuation_changes)
+                                .await?
+                        }
+                    }
+                }
+                None => {
+                    let message = format!("actuation provider for vss_id {} not available", vss_id);
```

Comment:
In most other error message we start with capital letter, ie. `"Actuation provider ..."`.